getting-started/gce/startup-script.sh

```diff
@@ -0,0 +1,66 @@
+#! /bin/sh
+
+# Copyright 2019 Google LLC
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START getting_started_gce_startup_script]
+set -ex
+
+# Talk to the metadata server to get the project id
+export GOOGLE_CLOUD_PROJECT=$(curl -s "http://metadata.google.internal/computeMetadata/v1/project/project-id" -H "Metadata-Flavor: Google")
+echo "Project ID: ${GOOGLE_CLOUD_PROJECT}"
+
+# Install logging monitor. The monitor will automatically pickup logs sent to syslog.
+curl -s "https://storage.googleapis.com/signals-agents/logging/google-fluentd-install.sh" | bash
+service google-fluentd restart &
+
+# Install Go and git with apt-get.
+apt-get install -yq ca-certificates git software-properties-common
+add-apt-repository -y ppa:longsleep/golang-backports
+apt-get install -yq golang-go
```

Comment:
I'd not expect users to install go toolchain on production machines. They build the binary on an environment and only will deploy the final binary.

Reply:
+1, I remember writing or reviewing some sample that had a separate script to build and upload to GCS, then the GCE startup script pulled the binary from GCS.

If that's too much indirection for a sample like this, then maybe separating into two scripts could work out better.

Or, if the actual code doesn't matter much, then we can upload a pre-built binary for them.

Reply:
Switched to use Cloud Build to build, tar, and upload to GCS. Seems better than a one-off script.

I also moved the systemd service file to a standalone file that gets included in the tar file.

That has the added benefit of making it clear where to add static assets later if needed.

Reply:
Nice, that sounds great.